edx_event_bus_kafka/internal/producer.py

```diff
@@ -210,6 +220,16 @@ def send(
         # poll(0) on a regular interval (see `poll_indefinitely`).
         self.producer.poll(0)
 
+    def _send_to_error_topic(self, *, topic: str, event_data: dict, event_key = None):
+        try:
+            event_key_as_str = str(event_key if event_key else "Missing Key")
+            event_data_as_str = json.dumps(event_data, allow_nan=True, skipkeys=True, default=lambda x: str(x))
+            self.producer.produce(f"{topic}-error", key=event_key_as_str, value=event_data_as_str)
```

Comment:
I think we would need to encode to bytes, here. (We don't actually have to worry about schemas for the error topic, right? This wouldn't be an SerializingProducer and wouldn't know about Avro.)

Reply:
It's already a regular (non-serializing) `Producer`, though -- I completed the work to separate out the serialization step. So all we need to do is convert the strings to bytes.

Reply:
Hm, it's interesting that it worked without going down to bytes. It's still probably a better idea, but now we know we can send other things.

Reply:
Ah, here we go -- looks like it accepts either string or bytes: https://docs.confluent.io/platform/current/clients/confluent-kafka-python/html/index.html#confluent_kafka.Producer.produce

Reply:
Do you think there's a value to one over the other? I can't see a particular advantage but at this point you know more about the guts of kafka than I do

Reply:
Probably doesn't matter -- some code in there is automatically encoding strings to bytes. Doing the encoding ourselves would be more explicit and so I'd favor it slightly, but I don't think it matters much either way.